pandas/tests/copy_view/test_methods.py

```diff
@@ -575,6 +576,48 @@ def test_sort_index(using_copy_on_write):
     tm.assert_series_equal(ser, ser_orig)
 
 
+@pytest.mark.parametrize(
+    "obj, kwargs", [(Series([1, 2, 3]), {}), (DataFrame({"a": [1, 2, 3]}), {"by": "a"})]
+)
+def test_sort_values(using_copy_on_write, obj, kwargs):
+    obj_orig = obj.copy()
+    obj2 = obj.sort_values(**kwargs)
+
+    if using_copy_on_write:
+        assert np.shares_memory(obj.values, obj2.values)
+    else:
+        assert not np.shares_memory(obj.values, obj2.values)
+
+    # mutating df triggers a copy-on-write for the column / block
+    obj2.iloc[0] = 0
+    assert not np.shares_memory(obj2.values, obj.values)
+    tm.assert_equal(obj, obj_orig)
+
+
+@pytest.mark.parametrize(
+    "obj, kwargs", [(Series([1, 2, 3]), {}), (DataFrame({"a": [1, 2, 3]}), {"by": "a"})]
+)
+def test_sort_values_inplace(using_copy_on_write, obj, kwargs, using_array_manager):
+    obj_orig = obj.copy()
+    view = obj[:]
+    obj.sort_values(inplace=True, **kwargs)
+
+    assert np.shares_memory(obj.values, view.values)
```

Comment:
Actually, this is failing with ArrayManager (failing CI)

Reply:
`.values` on a DataFrame is always a copy for ArrayManager, so the generic `obj.values` doesn't work here:

```suggestion
    if isinstance(obj, DataFrame):
        assert np.shares_memory(get_array(obj, "a"), get_array(view, "a"))
    else:
        assert np.shares_memory(obj.values, view.values)
```

Reply:
You will have to use the same pattern below, so can probably simplify this if we give the Series a name so can use `get_array` in all cases.

Reply:
While testing this, I just pushed the changes I was doing locally.